### PR TITLE
ci: publish cucumber reports as GitHub checks

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Publish GitHub checks with the cucumber report

## Why is it important?

Developers wont' need to go anywhere else but look at the GitHub checks.

It uses https://github.com/elastic/apm-pipeline-library/blob/main/vars/githubCheck.groovy

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -598,9 +598,38 @@ def generateFunctionalTestStep(Map args = [:]){
...
     }
   }
 }
+
+/**
+ Notify the GitHub check for the cucumber report
+ **/
+def githubCheckNotify(name, status, cucumberFile) {
+  def description = "${name} ${status.toLowerCase()}"
+  def context = name
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:618]:
```suggestion
  def checkName = name
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -598,9 +598,38 @@ def generateFunctionalTestStep(Map args = [:]){
...
+  def url = "${env.RUN_DISPLAY_URL}"
+  def repo = env.REPO
+  def sha = env.GIT_BASE_COMMIT
+  if (params.GITHUB_CHECK_NAME?.trim() && params.GITHUB_CHECK_REPO?.trim() && params.GITHUB_CHECK_SHA1?.trim()) {
+    context = "${params.GITHUB_CHECK_NAME}-cucumber"
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:623]:
```suggestion
    checkName = "${params.GITHUB_CHECK_NAME}-cucumber"
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -598,9 +598,38 @@ def generateFunctionalTestStep(Map args = [:]){
...
+    description = "${params.GITHUB_CHECK_NAME} ${status.toLowerCase()}"
+    sha = params.GITHUB_CHECK_SHA1
+    repo = params.GITHUB_CHECK_REPO
+  }
+  githubCheck(name: context,
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:628]:
```suggestion
  githubCheck(name: checkName,
```

**Review by mdelapenya — APPROVED** [review]
Great! Only a few comments about variable names. Not blockers at all

**Comment by v1v** [comment]
I wonder if we need to ask for the cucumber plugin then this particular implementation will be much simpler?

**Comment by mdelapenya** [comment]
It works! Although the html+css is not properly rendered, but it's good progress!

**Comment by mdelapenya** [comment]
> I wonder if we need to ask for the cucumber plugin then this particular implementation will be much simpler?

Reading their docs, not sure if that would work: https://plugins.jenkins.io/cucumber-reports/#plugin-content-use

> You must use a Freestyle project type in jenkins.

**Comment by v1v** [comment]
It supports pipelines too -> https://plugins.jenkins.io/cucumber-reports/#plugin-content-pipeline-usage

If that's the case and the URL predictable, we could create a GitHub status commit with a link to the Cucumber report

**Pull request closed without merging**